cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -659,6 +662,20 @@ def calculate_point_time(self, start_pos, end_pos, default_vel, default_acc):
 			point_time = 3.0	# use default point_time
 		return point_time
 
+	def calculate_point_velocities(self, prev, curr, post):
+		# set zero velocities for last trajectory point only
+		if not post:
+			return [0]*len(curr.positions)
+		else:
+			# check sign change
+			if prev and not numpy.all(numpy.equal(numpy.sign(numpy.subtract(numpy.array(curr.positions), numpy.array(prev.positions))), numpy.sign(numpy.subtract(numpy.array(post.positions), numpy.array(curr.positions))))):
```

Comment:
A bi of reformatting would make this more readable and thus understandable.

Reply:
Also, Instead of `numpy.subtract` you should be able to do just `numpy.array(post.positions) - numpy.array(curr.positions))`. According to https://docs.scipy.org/doc/numpy/reference/generated/numpy.subtract.html it should be equivalent for element-wise operations which is what we want here AFAIK